File: January 2023/11 January 2023/Python/Make array elements unique.py
class Solution:
    def minIncrements(self, arr, N):

        ans = 0
        arr.sort()
        for i in range(1,N):
            if(arr[i]<=arr[i-1]): 
                ans+=(arr[i-1]-arr[i]+1)
                arr[i]=arr[i-1]+1
        return ans

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    
    T = int(input())
    while T > 0: 
        N=int(input())
        arr = [int(i) for i in input().split()]
        ob = Solution()
        print(ob.minIncrements(arr,N))
        
        T -= 1
# } Driver Code Ends

File: January 2023/11 January 2023/C++/Make array elements unique.cpp
//{ Driver Code Starts
// Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends

class Solution {
  public:
    long long int minIncrements(vector<int> arr, int N) {

        long long int ans = 0;
        sort(arr.begin(),arr.end()); 
        for(int i=1; i<N;i++){
            if(arr[i]<=arr[i-1]){
                ans+=(arr[i-1]-arr[i]+1);
                arr[i]=arr[i-1]+1;
            }
        }
        return ans;
    
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int N;
        cin >> N;
        vector<int> arr(N);
        for (int i = 0; i < N; i++) {
            cin >> arr[i];
        }
        Solution obj;
        cout << obj.minIncrements(arr, N) << endl;
    }
}
// } Driver Code Ends